### PR TITLE
Add core WG notes from 2021-07-02

### PR DESCRIPTION
**Pull request by jrvanwhy**
[Rendered](https://github.com/jrvanwhy/tock/blob/core-notes-2021-07-02/doc/wg/core/notes/core-notes-2021-07-02.md)

These meeting notes are from 9 days ago, not last Friday (sorry for the delay).

### CODE REVIEW

File: doc/wg/core/notes/core-notes-2021-07-02.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  stuff settles. The high-level takeaway for the PR is we have a lot of
+  high-level functionality in methods on `Grant` that are generic over the type
+  in the Grant and/or closure. E.g. every time you call `Grant::enter` you
+  duplicate the logic in the binary. It's generally inlined into the use site,
+  and for boards like Imix that means 50 or more copies of the identical code.
+  This PR moves the portions of the functions that do not rely on the generic
+  parameters into non-generic functions that are marked `#[inline(never)]`. I
+  then call those from the generic functions. For code that doesn't need to be
+  duplicated for the generic types, you only have a single instance of those
+  function. As written, this saved about 7300 bytes on Imix. Functionallity
```
Comment (hudson-ayers) [inline doc/wg/core/notes/core-notes-2021-07-02.md:283]:
```suggestion
  function. As written, this saved about 7300 bytes on Imix. Functionally it is
```

**Review by hudson-ayers — DISMISSED** [review]

File: doc/wg/core/notes/core-notes-2021-07-02.md (hunk 1 above)
Comment (jrvanwhy) [reply doc/wg/core/notes/core-notes-2021-07-02.md]: Fixed.

(Sorry everyone for the amend + force-push, that was accidental).

File: doc/wg/core/notes/core-notes-2021-07-02.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+* Johnathan: Unfortunately this only applies to RISC-V, at least with the
+  current plans. On behalf of OpenTitan, LowRISC is working on an embedded
+  position-independent-code spec for RISC-V that they're contributing to LLVM
+  for both `clang` and `rustc`.
+* Amit: That's awesome. Do we know if it's going to support the kind of
+  PIC we rely on right now in ARM for C?
+* Johnathan: The current plans are such that it will initially look a lot like
+  that, but then it will change to be somewhat more efficient and eliminate the
+  global offset table. It was specifically designed for Tock's usage.
+* Amit: That would be huge, that's great.
```
Comment (brghena) [inline doc/wg/core/notes/core-notes-2021-07-02.md:25]: This is incredibly exciting. All the more so if it's being designed for OpenTitan explicitly. My memory from when we were first exploring this way back when is that ARM PIC does a bunch of little inefficient things that are unnecessary from a Tock perspective. The GOT being the biggest example.

**Review by hudson-ayers — APPROVED** [review]

**Comment by lschuermann** [comment]
Netlify seems to have its share of issues in the past few days, trying again given https://www.netlifystatus.com/ says it's a go.

bors r+

**Comment by lschuermann** [comment]
I just confirmed these builds run fine for me locally including `make alldoc`, so this is (as indicated in Netlify's) logs most likely unrelated to these changes.

@hudson-ayers @bradjc @ppannuto I can't merge manually. Can you merge this?

**Pull request merged**